src/components/Authentication/Signup.jsx

```diff
@@ -0,0 +1,109 @@
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import {
+  Box,
+  Button,
+  FormControl,
+  FormLabel,
+  Input,
+  VStack,
+  Heading,
+  Text,
+  Link,
+  useToast,
+} from '@chakra-ui/react';
+import { useAuth } from '../../contexts/AuthContext';
+import PropTypes from 'prop-types'; // Import PropTypes
+
+const Signup = ({ isAdmin }) => {
+  // React states for input fields
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const { signup } = useAuth();
+  // TODO: Setup Error Alert
+  // eslint-disable-next-line no-unused-vars
+  const [error, setError] = useState('');
+  const [loading, setLoading] = useState(false);
+  const toast = useToast();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    toast({
+      title: 'Login Failed',
+      description: error,
+      status: 'error',
+      duration: 3000,
+      isClosable: true,
+    });
+  }, [error]);
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has a missing dependency: 'toast'. Either include it or remove the dependency array.